RedBeanPHP/Repository/Fluid.php

```diff
@@ -140,6 +140,7 @@ private function modifySchema( OODBBean $bean, $property, $value, &$columns = NU
 						$bean->setMeta("sys.typeof.{$destinationColumnNoQ}", NULL);
 					}
 				}
+				$this->flushKeyMapsCache();
```

Comment:
While this seems to be the good place to do it. It still leads to some interesting situations:


     $Bob   = R::dispense(array('_type'=>'person', 'name'=>'Bob'));
     $Sarah = R::dispense(array('_type'=>'pupil', 'name'=>'Sarah'));
     $project = R::dispense('project');
     $project->student = $Bob;
     $id = R::store($project);
     $project = R::load('project', $id);
     var_dump( $project->student->name ); //Bob, as expected
     $project2 = R::dispense('project');
     $project2->student = $Sarah;
     //R::trash( $project ); -- this will give Sarah 2nd time?
    $id2 = R::store($project2);
    $project2 = R::load('project', $id2);
    var_dump( $project2->student->name ); //Bob?? No flush? Which key takes precedence?

Reply:
Well, changing the type of bean from `person` to `pupil` isn't triggering a FK change on the database so.... it's still getting the information from the `person` table.